### DISCUSSION TITLE
MQTT.js import error: This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.

### DISCUSSION THREAD

**Discussion by vmasdani** [denoland/deno#10105] [Q&A] [answered]
hi. I tried importing MQTT.js with a single js file [mqtt.min.js](https://unpkg.com/mqtt@4.2.6/dist/mqtt.min.js) like this, and tried to run and got this errror.
![Screenshot from 2021-04-10 08-50-31](https://user-images.githubusercontent.com/37009464/114254531-2b1a5b80-99da-11eb-9578-ef4bb9f90d57.png)
This is the code.
![Screenshot from 2021-04-10 08-50-15](https://user-images.githubusercontent.com/37009464/114254544-3a99a480-99da-11eb-892f-96eb725cd6ac.png)
Any idea or workaround to fix this? This is my deno version
![Screenshot from 2021-04-10 08-54-01](https://user-images.githubusercontent.com/37009464/114254560-556c1900-99da-11eb-91ae-662dc2f4b513.png)

**Comment by josephrocca** [comment, marked as answer]
I'm guessing this is because `MQTT.js` contains a browserified version of Node.js's `Buffer`, and that package tries to change prototypes using `__proto__`. This doesn't work because, unlike browsers, Deno doesn't allow `__proto__` to be overwritten ([relevant issue](https://github.com/denoland/deno/issues/4324)). One way to fix this (as mentioned [here](https://github.com/denoland/deno/issues/10222)) is to replace all instances of `obj.__proto__ = newPrototype` with [`Object.setPrototypeOf(obj, newPrototype)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf).

But actually, it looks like [the latest version](https://github.com/feross/buffer/blob/795bbb5bda1b39f1370ebd784bea6107b087e3a7/index.js#L54-L55) of browserify's Buffer uses `setPrototypeOf` instead of `__proto__`, so you may just be able to ask the maintainer of https://github.com/mqttjs/MQTT.js to re-browserify the package and publish a new version.

**Comment by benatkin** [comment]
I'm getting this with babel-standalone:

##### `build.js`

```ts
const example = `
import { createSignal, onCleanup } from "solid-js";
import { render } from "solid-js/web";

const CountingComponent = () => {
	const [count, setCount] = createSignal(0);
	const interval = setInterval(
		() => setCount(c => c + 1),
		1000
	);
	onCleanup(() => clearInterval(interval));
	return <div>Count value is {count()}</div>;
};

render(() => <CountingComponent />, document.getElementById("app"));
`;

import Babel from "https://esm.sh/@babel/standalone";
import solid from "https://esm.sh/babel-preset-solid";
Babel.registerPreset("solid", solid());
console.log(Babel.transform(example, {presets: ["solid"]}).code);
```

```
❯ deno run --allow-env=BABEL_TYPES_8_BREAKING build.js
This browser lacks typed array (Uint8Array) support which is required by `buffer` v5.x. Use `buffer` v4.x if you require old browser support.
import { template as _$template } from "solid-js/web";
import { createComponent as _$createComponent } from "solid-js/web";
import { insert as _$insert } from "solid-js/web";

const _tmpl$ = _$template(`<div>Count value is </div>`, 2);

import { createSignal, onCleanup } from "solid-js";
import { render } from "solid-js/web";

const CountingComponent = () => {
  const [count, setCount] = createSignal(0);
  const interval = setInterval(() => setCount(c => c + 1), 1000);
  onCleanup(() => clearInterval(interval));
  return (() => {
    const _el$ = _tmpl$.cloneNode(true),
          _el$2 = _el$.firstChild;

    _$insert(_el$, count, null);

    return _el$;
  })();
};

render(() => _$createComponent(CountingComponent, {}), document.getElementById("app"));
```
